src/amo/components/CollectionAddAddon/index.js

```diff
@@ -105,31 +139,33 @@ export class CollectionAddAddonBase extends React.Component<InternalProps> {
         username: collection.authorUsername,
       }),
     );
-    this.props.setUIState({ addonWasAdded: false });
-  };
-
-  resetMessageStatus = () => {
-    this.props.setUIState({
-      addonWasAdded: false,
-    });
+    this.resetMessageStatus();
   };
 
   render() {
     const { errorHandler, i18n, uiState } = this.props;
 
     return (
-      <div className="CollectionAddAddon">
+      <Card className="CollectionAddAddon">
         {errorHandler.hasError() ? (
           errorHandler.renderError()
         ) : (
           <TransitionGroup className="CollectionAddAddon-noticePlaceholder">
-            {uiState.addonWasAdded && (
+            {uiState.completedAddonAction && (
               <CSSTransition
                 classNames="CollectionAddAddon-noticePlaceholder-transition"
                 timeout={MESSAGE_FADEOUT_TIME}
               >
-                <Notice type="success">
-                  {i18n.gettext('Added to collection')}
+                <Notice
+                  type={
+                    uiState.completedAddonAction === addonAddedAction
+                      ? 'success'
+                      : 'generic'
+                  }
+                >
+                  {uiState.completedAddonAction === addonAddedAction
```

Comment:
can we extract out completedAddonAction since it's used a few times? 

Also/or maybe this itself could be a var `const isAdded = uiState.completedAddonAction === addonAddedAction`